mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -1,29 +1,52 @@
 import re
 import csv
+import logging
 
-from mfr.extensions.tabular.exceptions import EmptyTableError, TabularRendererError
 from mfr.extensions.tabular import utilities
+from mfr.extensions.tabular.settings import MAX_FILE_SIZE, TABULAR_INIT_SNIFF_SIZE
+from mfr.extensions.tabular.exceptions import EmptyTableError, TabularRendererError
+
+logger = logging.getLogger(__name__)
 
 
 def csv_stdlib(fp):
     """Read and convert a csv file to JSON format using the python standard library
-    :param fp: File pointer object
-    :return: tuple of table headers and data
+
+    Quirk: ``csv.Sniffer().sniff()`` needs the FULL first row and ONLY one full row to be able to
+    effectively detect the correct dialect of the file.
+
+    :param fp: the file pointer object
+    :return: a tuple of table headers and data
     """
-    data = fp.read(2048)
+
+    logger.info('>>> ??? &&& ~~~ current settings')
+    logger.info('>>> ??? &&& ~~~ max rendering file size = {}'.format(MAX_FILE_SIZE))
+    logger.info('>>> ??? &&& ~~~ initial sniffing size = {}'.format(TABULAR_INIT_SNIFF_SIZE))
+
+    # Prepare the first row for sniffing
+    data = fp.read(TABULAR_INIT_SNIFF_SIZE)
+    data = _trim_or_append_data(fp, data, TABULAR_INIT_SNIFF_SIZE, 0)
+
+    # Reset the file pointer
```

Comment:
The first solution (see below) didn't work 100% since for one file, sniffing the full file ended up in wrong delimiter while sniffing only the first row worked as expected. This is why even the sniff can sniff the full file, I only provide it with the first row.

```python
...
data = fp.read(TABULAR_INIT_SNIFF_SIZE)
if len(data) == TABULAR_INIT_SNIFF_SIZE:
    data = _trim_or_append_data(fp, data, TABULAR_INIT_SNIFF_SIZE, 0)
...
```